### PR TITLE
UI-1220 Fix BadgeList with no items

### PR DESCRIPTION
**Pull request by hanzge**
- Modify `items` propType
- return `null` when `items` is empty
- Fix test case and add one storybook example

### CODE REVIEW

File: lib/components/Badge/BadgeList.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const propTypes = {
   /**
    * Array of Badges
    */
-  items: PropTypes.array,
+  items: PropTypes.arrayOf(PropTypes.object),
```
Comment (hanzge) [inline lib/components/Badge/BadgeList.js]: I use `PropTypes.object` because the shape of object can be anything. As long as developer passes `labelKey`, `items` can accept different shapes of data

File: lib/components/Badge/BadgeList.js [hunk 2]
```diff
@@ -56,6 +56,10 @@ const BadgeList = ({ items, className, isRemovable, isStacked, labelKey, onRemov
   if (isStacked && isRemovable) {
     throw new Error('Stacked BadgeList items cannot be removable');
   }
+  if (items.length === 0) {
```
Comment (mallik-svmx) [inline lib/components/Badge/BadgeList.js]: !items.length

File: lib/components/Badge/BadgeList.test.js [hunk 3]
```diff
@@ -4,8 +4,8 @@ import { shallow } from 'enzyme';
 import BadgeList from './BadgeList';
 
 describe('BadgeList', () => {
-  it('should render badge with className', () => {
+  it('should render nothing when no items are passed', () => {
```
Comment (mallik-svmx) [inline lib/components/Badge/BadgeList.test.js]: test description and expectations seems irrelevant, may expect children length/
